rs/pocket_ic_server/src/state_api/state.rs

```diff
@@ -709,12 +709,21 @@ impl ApiState {
         self.stop_progress(instance_id).await;
         let instances = self.instances.read().await;
         let mut canister_http_adapters = self.canister_http_adapters.write().await;
-        canister_http_adapters.remove(&instance_id);
-        let mut instance_state = instances[instance_id].lock().await;
-        if let InstanceState::Available(pocket_ic) =
-            std::mem::replace(&mut *instance_state, InstanceState::Deleted)
-        {
-            std::mem::drop(pocket_ic);
+        loop {
+            let mut instance_state = instances[instance_id].lock().await;
+            match std::mem::replace(&mut *instance_state, InstanceState::Deleted) {
+                InstanceState::Available(pocket_ic) => {
+                    std::mem::drop(pocket_ic);
+                    canister_http_adapters.remove(&instance_id);
+                    break;
+                }
+                InstanceState::Deleted => {
+                    break;
+                }
+                InstanceState::Busy { .. } => {}
+            }
+            drop(instance_state);
+            tokio::time::sleep(Duration::from_secs(1)).await;
```

Comment:
In the meantime, the instance could come out of `Busy` and enter `Busy` all over again, correct? So this is a best-effort kind of loop, and it might do quite a lot of rounds if the instance is being bombarded by requests. 

But then of course, this scenario does not make sense. Why would a user drop an instance and continue to send requests to it.

Reply:
> In the meantime, the instance could come out of Busy and enter Busy all over again, correct? So this is a best-effort kind of loop, and it might do quite a lot of rounds if the instance is being bombarded by requests.

Indeed.

> But then, this scenario does not make sense. Why would a user drop an instance and continue to send requests to it.

Imagine that dfx receives CTRL-C and thus wants to delete the instance while the user still runs a test sending requests to that instance.